x-pack/test/functional/services/ml/field_stats_flyout.ts

```diff
@@ -0,0 +1,138 @@
+      fieldName: string,
+      fieldType?: 'keyword' | 'date' | 'number'
+    ) {
+      await retry.tryForTime(2000, async () => {
+        // escape popover
+        await browser.pressKeys(browser.keys.ESCAPE);
+
+        if (fieldType === 'date') {
+          await testSubjects.existOrFail(`mlFieldStatsFlyoutContent ${fieldName}-histogram`);
+        }
+
+        if (fieldType === 'keyword') {
+          await testSubjects.existOrFail(`mlFieldStatsFlyoutContent ${fieldName}-topValues`);
+        }
+
+        // For numeric fields, we expect both the top values and the distribution chart
+        if (fieldType === 'number') {
+          // Assert top values exist
+          await testSubjects.existOrFail(
+            `mlFieldStatsFlyoutContent ${fieldName}-buttonGroup-topValuesButton`
+          );
+          await testSubjects.click(
+            `mlFieldStatsFlyoutContent ${fieldName}-buttonGroup-topValuesButton`
+          );
+
+          await testSubjects.existOrFail(`mlFieldStatsFlyoutContent ${fieldName}-topValues`);
+
+          // Assert distribution chart exists
+          await testSubjects.existOrFail(
+            `mlFieldStatsFlyoutContent ${fieldName}-buttonGroup-distributionButton`
+          );
+          await testSubjects.click(
+            `mlFieldStatsFlyoutContent ${fieldName}-buttonGroup-distributionButton`
+          );
+          expect(
+            await find.existsByCssSelector('[data-test-subj="mlFieldStatsFlyout"] .echChart')
+          ).to.eql(true);
+        }
+      });
+    },
+
+    async clickFieldStatTrigger(
+      testSubj: string,
+      fieldName: string,
+      fieldType?: 'keyword' | 'date' | 'number'
+    ) {
+      const selector = `~${testSubj} > ~mlInspectFieldStatsButton-${fieldName}`;
+
+      await retry.tryForTime(2000, async () => {
+        await testSubjects.existOrFail(selector);
+        await testSubjects.click(selector);
+        await testSubjects.existOrFail('mlFieldStatsFlyout');
+        await testSubjects.existOrFail(`mlFieldStatsFlyoutContent ${fieldName}-title`);
+      });
+      await this.assertFieldStatContentByType(testSubj, fieldName, fieldType);
+    },
+    async clickFieldStatComboBoxTrigger(
+      parentComboBoxSelector: string,
+      fieldName: string,
+      fieldType?: 'keyword' | 'date' | 'number'
+    ) {
+      const selector = `mlInspectFieldStatsButton-${fieldName}`;
+
+      await retry.tryForTime(2000, async () => {
+        const fieldTarget = await testSubjects.find(parentComboBoxSelector);
+        await comboBox.openOptionsList(fieldTarget);
+
+        await testSubjects.existOrFail(selector);
+        await testSubjects.click(selector);
+        await testSubjects.existOrFail('mlFieldStatsFlyout');
+
+        await testSubjects.existOrFail(`mlFieldStatsFlyoutContent ${fieldName}-title`);
+
+        await this.assertFieldStatContentByType(parentComboBoxSelector, fieldName, fieldType);
+        await this.ensureFieldStatsFlyoutClosed();
```

Comment:
Putting the flyout close at this service method level saves us a lot of extra lines in the test files. But there are two things to mention IMO:
1. we probably also want to add it to `clickFieldStatTrigger`
2. the method naming can be confusing: `clickFieldStatComboBoxTrigger` / `clickFieldStatTrigger` are actually opening the flyout, asserting content and closing it again, which we can't really derive from the name. If we want to keep it that way, I'd suggest to rename these service methods (along with their wrappers in transform and DFA services) for clarity away from `clickXYZ` to `assertXYZ` or `checkXYZ`.

Reply:
@pheyos You're right that the naming can be confusing. I've renamed them to assertions here [`10d9917` (#150927)](https://github.com/elastic/kibana/pull/150927/commits/10d99177e34f3ff4ad659ce873377445e155deb4)